backends/ebpf/ebpfParser.cpp

```diff
@@ -73,6 +74,49 @@ StateTranslationVisitor::compileLookahead(const IR::Expression* destination) {
     builder->blockEnd(true);
 }
 
+void
+StateTranslationVisitor::compileAdvance(const P4::ExternMethod* extMethod) {
+    auto argExpr = extMethod->expr->arguments->at(0)->expression;
+    cstring argStr;
+    if (auto cnst = argExpr->to<IR::Constant>()) {
+        argStr = cstring::to_cstring(cnst->asUnsigned());
+    } else {
+        argStr = argExpr->toString();
```

Comment:
I don't think this can be correct for an arbitrary expression argExpr. This will generate a P4 representation of the expression, which has nothing to do with the C code you need to generate. You need to call the code generator for expressions here.

Reply:
This was just for logging purposes. We don't have `emitTraceMessage` method that would receive expression as an argument.  
So the easiest way was to put there P4 representation (for logging I thougth it might be enough). 
My proposition is to emit a trace message only in case of Constant. Do you agree?